src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -629,6 +652,65 @@ FlightContext resume(String stairwayId, String flightId)
     return null;
   }
 
+  FlightContext makeFlightContextById(String flightId)
+      throws DatabaseOperationException, InterruptedException {
+    final String sqlGetFlight =
+        "SELECT class_name, debug_info, status "
+            + " FROM "
+            + FLIGHT_TABLE
+            + " WHERE flightid = :flightId";
+
+    for (int retry = 0; retry < SERIALIZATION_RETRIES; retry++) {
+      try (Connection connection = dataSource.getConnection();
+          NamedParameterPreparedStatement getFlightStatement =
+              new NamedParameterPreparedStatement(connection, sqlGetFlight)) {
+
+        startTransaction(connection);
+
+        getFlightStatement.setString("flightId", flightId);
+        FlightContext flightContext = null;
+        try (ResultSet rs = getFlightStatement.getPreparedStatement().executeQuery()) {
+          if (rs.next()) {
+            flightContext = makeFlightContext(connection, flightId, rs);
+          }
```

Comment:
No. We commit the transaction and return null.

Reply:
In that case a NPE will be triggered, since the next line after the call to this method passes the `null` result to `queueFlight()` which calls `flightContext.getFlightId()`.